File: spearmint/src/node.rs
#[derive(Hash, PartialEq, Eq, Debug)]
pub struct Node {
    pub branch: &'static str,
    pub commit: &'static str,
    pub parent: Option<&'static str>,
    pub children: Vec<&'static str>,
    pub is_owned: bool,
    pub commit_prefix: &'static str,
    pub commit_suffix: &'static str,
}

impl Node {
    pub fn new (
        branch: &'static str,
        commit: &'static str,
        parent: Option<&'static str>,
        children: Vec<&'static str>,
        is_owned: bool,
        commit_prefix: &'static str,
        commit_suffix: &'static str) -> Self {
            Node { branch, commit, parent, children, is_owned, commit_prefix, commit_suffix }
        }
}

File: src/repository/node.py
# Copyright 2022 The Gum Authors
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

from dataclasses import dataclass


@dataclass
class Node:
    # The branch name eg. "aaaaa".
    branch: str
    # The full commit hash.
    commit: str
    # The parent commit node.
    parent: "Node"
    # List of child commit nodes.
    children: list["Node"]
    # Boolean denoting whether the change is owned.
    is_owned: bool
    # Shortest unique commit hash prefix.
    commitPrefix: str
    # Remainder of the shortened commit hash.
    commitSuffix: str
